Assets/Editor Toolbox/Runtime/Serialization/SceneSerializationUtility.cs

```diff
@@ -44,6 +44,10 @@ private static void RefreshCache()
             foreach (var scene in EditorBuildSettings.scenes)
             {
                 buildIndex++;
+
+                if (string.IsNullOrEmpty(scene.path))
```

Comment:
You should do it before incrementing the buildIndex, additionally please add brackets to follow the project conventions
```
if (string.IsNullOrEmpty(scene.path))
{

}
```

Reply:
It is done after the build index so it can carry on with other vaild scene added in the build settings, so this is as intended addon.

I am not sure what the class exactly does so unless it should not refresh other scene caches if one scene is not vaild, I will make the changes.

Reply:
To calculate properly build indexes you should skip this incrementation, same way Unity displays indexes in the Build Settings, e.g.
```
- Sample Scene [0]
- <none> [-]
- Sample Scene 2 [1]
```

where **[x]** is the current index.

Reply:
Ah i see, I did not notice it skip the index for the deleted scene.